### ISSUE TITLE
hottclasses utf8 notations are incompatible with main library

### ISSUE THREAD

**Issue by Alizter** [HoTT/HoTT#1140]
Observe the following:

```coq
Require Import Basics.
Require Import Basics.Utf8.
Require Import Classes.interfaces.canonical_names.

Notation "_ ⊔ _" is already defined at level 85 with arguments constr at next level,
constr at level 85 while it is now required to be at level 50 with arguments constr
at next level, constr at next level.
```

As far as I can tell, hottclasses is littered with utf8 notations (and other notations) declared at their own level. This makes using hottclasses with the rest of the library difficult.

I think we need to properly integrate hottclasses  with the rest of the library, that means reserving these notations in Basics/Utf8.v and properly adding scope to all these notations.

I know that a lot of files in hottclasses use utf8 in their code. Our style guidelines currently discourage use of utf8. I think it might be time to open this discussion again since we don't have an unlimited amount of ascii symbols.

In the group theory library (in Algebra), I want to be able to allow people to use utf8 notations for isomorphisms and direct products. I am currently trying to get rid of all utf8 that I used, but leaving in the option.

Jason's category theory library seperates utf8 notations really well. I think we should follow his example.

**Comment by mikeshulman** [comment]
The reason we decided against using unicode by default was not just some arbitrary dislike of it, but the fact that some installations actually had concrete problems with it.  Unless we can verify that those problems have gone away, I wouldn't be in favor of requiring all library users to use unicode.

Can you say more specifically what the category theory library does that you'd like to emulate?

**Comment by Alizter** [comment]
In Categories/Category/Core.v there is a module with the (normal) notations defined in it. Then other files can import these notations if they wish. I was proposing we do this for the unicode notations, but I see now that might not be possible if the problems with unicode persist.

I will search for what these concrete problems were and what we can do about them.

**Comment by Alizter** [comment]
I propose we set out clear guidelines for using Utf8 notations. There is nothing about this in STYLE.md currently. We need to think about what the best technique is in keeping these separate enough.
 
I don't think there is any technical argument with unicode, the complaints in #377 seem to be user based. We should make it the default option not to use unicode. Some seem to say "lots of unicode symbols in code will scare mathematicians away". It is already the case that lots of symbols anywhere will scare people away in any context.

Supposedly if writing unicode in most text editors is a simple as latex, shouldn't this be something we take advantage of?

We can always ask people nicely to leave all the symbols in a comment in the bottom, so copy and past is always possible.

I can't find anything about unicode breaking peoples installation however.

**Comment by mikeshulman** [comment]
> Some seem to say "lots of unicode symbols in code will scare mathematicians away". It is already the case that lots of symbols anywhere will scare people away in any context.

No, it's different.  Mathematicians are comfortable with symbols in general.  Nearly all of them are by now comfortable with (La)TeX.  But Unicode is still fairly new, and

> Supposedly if writing unicode in most text editors is a simple as latex

I have yet to see convincing evidence of this.  Even in Emacs it's not totally straightforward; the built-in TeX input mode has pretty limited functionality, and adding new symbols to any input mode requires AFAIK some nontrivial lisp hackery.

**Comment by Alizter** [comment]
This is closed by #1424
